### DISCUSSION TITLE
xx% of predicted values are same（0-th prediction）！

### DISCUSSION THREAD

**Discussion by CuiZhengliang** [PaddlePaddle/PaddleSpatial#141] [BaiduKDDCup2022] [answered]
![image](https://user-images.githubusercontent.com/20610180/171784203-9ed33979-06e4-4bbd-9123-9c677d83f4ba.png)

I got this msg, when i submit. I have check my predict.py on local machine, results of predict has no problems. and I have submit success before. 

my team name:  Run

**Comment by noritoshitamura** [comment]
I met the error before.
In my case,   "checkpoints" of prepare.py was absent.

**Reply by CuiZhengliang** [reply]
thx, but i have params "checkpoints" of my prepare.py

**Comment by wang975679801** [comment]
I have the same problem. I do not think that predicting similar values is abnormal. For example, using the mean value of Patv in training dataset is a common baseline. It is even better than the open source GRU baseline. For some time periods that are hard to predict, using mean value to replace the model prediction is also a reasonable and robust strategy.

**Reply by CuiZhengliang** [reply]
I couldn't agree with you more. There is no weather data, but the most direct impact of wind power comes from the weather. Therefore, the model does not learn good rules, so the prediction is more robust when the fluctuation is small.

**Comment by ramber1836** [comment]
I do not think predicting future 288 points with the same (or over 90% identical) value is a practical solution, though the rmse/mae might be smaller. The prediction with small fluctuation does not mean 90%+ of predicted future points are the same. 

BTW, long-term time series forecasting is really challenging, esp. for Wind Power Forecasting, which, thus, motivates this competition.

**Reply by wang975679801** [reply]
Such judgments are too arbitrary. For example, there maybe a day with almost no wind. Then an ideal model will predict the patv=0 at all 288 points. My team name is SlienceG Team. We use multiple trained models to predict Patv with no heuristic operations. I think we should respect the judgement of models, but not to directly treat them as abnormal ones.

**Reply by wang975679801** [reply]
Once the prediction of just one turbine has 90% same values, the system will return error. It's too strict. It might be a better choice to check the similar values over all 134*288 prediction or change 90% to 99%.

**Reply by wang975679801** [reply]
Besides, in metrics.py, it first filter the predictions with reasonable raw data. And then check if the predictions are abnormal. I think it maybe better to check the ptedictions first, and then filter the raw data. Because, after filtering the raw data, there would be much less data points. Perhaps, some unexpected things happened.

**Reply by ramber1836** [reply]
> Once the prediction of just one turbine has 90% same values, the system will return error. It's too strict. It might be a better choice to check the similar values over all 134*288 prediction or change 90% to 99%.

Checking the similar prediction results of different turbines sounds reasonable. We will figure this out.

**Comment by ramber1836** [comment, marked as answer]
@CuiZhengliang @wang975679801 
According to the suggestions proposed by @wang975679801, the rules for checking identical predicted values in the online test have been updated.
Please refer to the latest metrics.py script for more details. 
Thanks.
